gpytorch/module.py

```diff
@@ -87,7 +87,11 @@ def initialize(self, **kwargs):
                 val = float(val)
             if "." in name:
                 module, name = self._get_module_and_name(name)
-                module.initialize(**{name: val})
+                if isinstance(module, nn.ModuleList):
+                    idx, name = name.split(".", 1)
```

Comment:
Should we handle the case of an empty `ModuleList` in the model (in that case you'd get a `TypeError` from trying to unpack a single element)? Might be safe to ignore since one generally wouldn't have that as a key in the first place...

Reply:
I was thinking that `ModuleLists` are organically created from composing kernels (and so wouldn't be empty) but sure it makes sense to support that. This also makes the assumption that `ModuleList`s contains `Modules` which can contain other `ModuleList`s but `ModuleList`s aren't directly nested.

Reply:
I smell recursion...

Reply:
Oops duh I wrote my test wrong; nested module lists will have a `kernels` in its namespace so it yep it will do the right thing. Will update with the empty ModuleList case.

Reply:
yeah I guess that's what I mean by 

> Might be safe to ignore since one generally wouldn't have that as a key in the first place...